.github/workflows/deployment.yml

```diff
@@ -1,4 +1,4 @@
-name: "deployment"
+name: "Deployment staging"
```

Comment:
Maybe you could reuse workflows to eliminate duplication: https://docs.github.com/en/actions/learn-github-actions/reusing-workflows ? It's a brand new feature.